File: CS1/cs prg/WindowsApplicationThread/Backup/WindowsApplicationThread/Class1.cs
using System;
using System.Collections.Generic;
using System.Text;

using System.Threading;
namespace WindowsApplicationThread
{
    public class Class1
    {
        private int Runs = 0;
       
        public void CountUp()
        {
            while (Runs < 10)
            {
             // Monitor.Enter(this);
                int Temp = Runs;
                Temp++;
                Console.WriteLine(Thread.CurrentThread.Name + " " + Temp);
                Thread.Sleep(1000);
                Runs = Temp;
              //Monitor.Exit(this);
            }
        }
        public void RunThreads()
        {
            Thread t1 = new Thread(new ThreadStart(CountUp));
            t1.Name = "t1";
            Thread t2 = new Thread(new ThreadStart(CountUp));
            t2.Name = "t2";

            t1.Start();
            t2.Start();

        }
    }
}

File: VB.NET/windows/WindowsThreadApplication/WindowsThreadApplication/Class1.vb
Imports System.Threading

Public Class Class1
    Private Runs As Integer = 0

    Public Sub CountUp()
        Do While (Runs < 10)
            Monitor.Enter(Me)
            Dim Temp As Integer = Runs
            Temp += 1
            Console.WriteLine(Thread.CurrentThread.Name & " " & Temp)
            Thread.Sleep(1000)
            Runs = Temp
            Monitor.Exit(Me)
        Loop
    End Sub
    Public Sub RunThreads()
        Dim t2 As Thread = New Thread(AddressOf CountUp)
        t2.Name = "t2"
        Dim t3 As Thread = New Thread(AddressOf CountUp)
        t3.Name = "t3"
        t2.Start()
        t3.Start()
    End Sub
End Class
		